File: Aufgabe_3/aufgabe3/Util/Time.php
<?php

/**
 * Created by PhpStorm.
 * User: dima
 * Date: 11.05.16
 * Time: 11:23
 */
class Time
{

    protected $timeInSec;

    public function __construct()
    {
        $this -> year   = date("Y");
        $this -> month  = date("n");
        $this -> day    = date("j");

        $this -> hour = date("H")+1;
        $this -> min  = date("i")+1;


        $this -> timeInSec =
            ($this->min  * 60)+
            ($this->hour * 60 * 60) +
            ($this->day  * 24 * 60 * 60) +
            ($this->month * 30 * 24 * 60 * 60) +
            ($this->year * 12 * 30 * 24 * 60 * 60);
    }


    public function getTimeInSec()
    {
        return $this->timeInSec;
    }

    public function getTimeInSecPlus30min()
    {
        $minInSec30 = 30*60;
        return $this->timeInSec + $minInSec30;
    }

}

?>

File: Aufgabe_3/adminUI_ITS/PHPAdmin/src/Time.java
import java.util.Calendar;

public class Time {

	private int year;
	private int month;
	private int day;
	private int hour; 
	private int min;
	private int timeout;
	
	private long minInSec;
	private long hourInSec;
	private long dayInSec;
	private long monthInSec;
	private long yearInSec;
	private long sumTime;
	
	public Time() {
		Calendar time = Calendar.getInstance();
	    year = time.get(Calendar.YEAR);
	    month = time.get(Calendar.MONTH) + 1;  //wegen 0 bis 11
	    day = time.get(Calendar.DAY_OF_MONTH);
	    hour= time.get(Calendar.HOUR_OF_DAY);
	    min = time.get(Calendar.MINUTE);
	    
	    timeout = 30 * 60;
	    
	    minInSec = min * 60L;
	    hourInSec = hour * 60L * 60L;
	    dayInSec = day * 24L * 60L * 60L;
	    monthInSec = month * 30L * 24L * 60L * 60L;
	    yearInSec = year * 12L * 30L * 24L * 60L * 60L;
	    
	    sumTime = yearInSec + monthInSec + dayInSec + hourInSec + minInSec;
	}
	
	public long getTimeInSecPlus30Min() {
		return sumTime + timeout;
		
	}
	
	public long getTimeInSec() {
		return sumTime;
	}
}
